crates/miden-tx/src/errors/mod.rs

```diff
@@ -27,19 +27,41 @@ use thiserror::Error;
 
 #[derive(Debug, Error)]
 pub enum NoteCheckerError {
+    #[error("invalid input note count {0} is out of range)")]
```

Comment:
```suggestion
    #[error("input note count {0} is out of range)")]
```

Nit: The invalid can probably be dropped?